### DISCUSSION TITLE
Help - Can't Create a New Project

### DISCUSSION THREAD

**Discussion by Bodyclock-Games** [stride3d/stride#1960] [Q&A]
Hi All. Am trying to create a new project. I started by trying to load the UI sample projects but they all failed with a version issue. I then tried creating a new project from the base template and I got this error on the startup screen along with operation cancelled. The folder and sub folders are created but no solution files.

I tried other templates with no success. Same error. I tried different locations and drives. I tried uninstalling and reinstalling both the launcher and engine, no joy. Any help would be appreciated. It's been a frustrating afternoon.

I don't know what file already exists but I've checked and can't find anything left behind.

I can open an existing project.

```
Verbose: Generating Terra...
Error: Unexpected exception while processing [Source: $ProjectName$.csproj.t4, Target: $ProjectName$.csproj, IsTemplate: True]
IOException: The file exists.
   at System.IO.Path.GetTempFileName()
   at Mono.TextTemplating.TemplatingEngine.CompileCode2(IEnumerable`1 references, TemplateSettings settings, CodeCompileUnit ccu)
   at Mono.TextTemplating.TemplatingEngine.CompileTemplateInternal(ParsedTemplate pt, String content, ITextTemplatingEngineHost host)
   at Mono.TextTemplating.TemplatingEngine.CompileTemplate(String content, ITextTemplatingEngineHost host)
   at Mono.TextTemplating.TemplatingEngine.ProcessTemplate(String content, ITextTemplatingEngineHost host)
   at Stride.Core.ProjectTemplating.ProjectTemplate.Generate(String outputDirectory, String projectName, Guid projectGuid, ILogger log, IReadOnlyDictionary`2 options, List`1 generatedOutputFiles) in C:\BuildAgent\work\b5f46e3c4829a09e\sources\tools\Stride.Core.ProjectTemplating\ProjectTemplate.cs:line 200
```

**Comment by ly3027929699** [comment]
it occured when i deleted the 6.0 sdk, as i works on  the branch of .net8

**Comment by Bodyclock-Games** [comment]
Thanks, I had a look at Visual Studio. I did a repair but no luck. I have the following installed.

.NET 6.0 Runtime (Long Term Support)
.NET 7.0 Runtime
.NET SDK

Edit: After opening an existing project VS forced the installation of the following:

.NET Framework 4.7.2 targeting pack
.NET Framework 4.8 SDK
.NET Framework 4.8 targeting pack

and other seeming unrelated packages. But still no joy.

Is there a seperate .NET 6.0 SDK? I can't see one that is available.
Plenty of .NET framework SDK's available.
Is there anything else that should be installed?

Any help appreciated.

Cheers

Chris

**Reply by IXLLEGACYIXL** [reply]
you can get your installed sdks by writing `dotnet --list-sdks` in powershell

there net6 has to appear else it wont work
https://dotnet.microsoft.com/en-us/download/dotnet/6.0
here you can get the net6 sdk

after installing it reboot your pc

**Comment by Bodyclock-Games** [comment]
Thanks. It seems I have the following SDK's installed. I installed the latest and restarted the machine but still the same error?

5.0.101 [C:\Program Files\dotnet\sdk]
6.0.414 [C:\Program Files\dotnet\sdk]
6.0.415 [C:\Program Files\dotnet\sdk]
7.0.401 [C:\Program Files\dotnet\sdk]

**Comment by Bodyclock-Games** [comment]
I just had a look at the build tools in Visual Studio and it seems I have none installed???? Could that be the problem?

Edit: But some of them where installed in the forced VS update when I opened an existing project.

**Comment by Bodyclock-Games** [comment]
Hi All, Just to give this a bump in the hopes for any insights. I am begining to think this isn't a Stride issue.
So I can run Stride with an existing project, so I assume all the .NET prerequisites are in place. I also assume the issue is the creating of a C# Visual Studio project. I have done a complete uninstall of VS 2022 and the problem still exists. Installed VS Code but the problem still exists.

A couple of days ago the where some updates for Visual Studio which I clicked to install. I got the Oops, something went wrong error. Didn't think much of it. I uninstalled VS 2022 and now I can't reinstall it (Oops, something went wrong - which is seeming less cute by the moment). I can't even report an error through VS Installer, nothing happens. And that appears to be the only way of reporting an error to Microsoft.

Anyway, it's been a frustrating couple of days. Have deleted more registry keys than ever before. My next step is to try VS 2019 (assuming I can still get an install).

So if anyone has any clues, it would be very much appreciated.

**Reply by IXLLEGACYIXL** [reply]
when you get an error its never a good idea to delete any registry keys
first run `DISM.exe /Online /Cleanup-image /Restorehealth` in admin powershell
then run `sfc /scannow`

then make a reboot of the pc ( not shutdown ) 
then uninstall everything related to VS

then try installing with admin rights

**Reply by Bodyclock-Games** [reply]
Thanks, I was just following the Microsoft advice. I'll retry using powershell.

**Reply by IXLLEGACYIXL** [reply]
cmd or powershell doesnt matter

**Reply by Bodyclock-Games** [reply]
Thanks for your efforts. All the command lines completed successfully but still no joy with a VS installation. The installer installs and then says Oops.

**Reply by IXLLEGACYIXL** [reply]
can you just install it from here, fresh downloaded?
https://visualstudio.microsoft.com/de/downloads/

**Reply by Bodyclock-Games** [reply]
Done all that, and 2019 as well, the installer still fails with the same error.

**Reply by IXLLEGACYIXL** [reply]
do you have an anti virus instlaled?

**Reply by Bodyclock-Games** [reply]
I thought of that, but I only have Windows Defender and I tried turning off the ransomware blocker which was recommended. That had no effect.

**Reply by IXLLEGACYIXL** [reply]
i would just reinstall windows :D

**Reply by Bodyclock-Games** [reply]
My last resort but that is such a pain :(

**Reply by IXLLEGACYIXL** [reply]
you could try to clena all the temp files wth data cleaner and try it again

**Comment by Doprez** [comment]
`The file exists.` does this specific error happen with any project? or is it only the UI sample?

EDIT: nevermind I read the shole thing.

**Reply by Doprez** [reply]
https://github.com/Doprez/stride-physics-demo can you pull down this project and see if it opens with Stride? Its a very simple demo of Strides Bullet physics so you can delete the few scripts from me and use this.

I'm assuming the issue is only with creating new projects with Stride but I just want to be sure.

**Reply by Bodyclock-Games** [reply]
It's only creating new projects in Stride. It seems to be unable to create a new solution file.

Anyway one Windows reinstall later and it's all working fine minus a load of apps to reinstall. Probably needed a cleanup anyway. Frustrating thing is not knowing the problem.

Thanks for all the help guys.

**Reply by Doprez** [reply]
Ahh that's too bad, sorry we couldnt find a less painful way.

**Reply by Doprez** [reply]
Just some added info in case some in the future finds this problem. The error stacktrace pointing at `ProjectTemplate.cs:line 200` doesnt make a lot of sense. Below is where it is throwing that error in source and thereis nothing at that line.
![image](https://github.com/stride3d/stride/assets/73259914/4501f5c6-9b01-40e4-bbce-e803b2bb3e61)

It **could** be a difference between release and debug builds of Stride but this is probably worth noting at least if someone else has this error.
